connection/auth.go

```diff
@@ -15,8 +15,10 @@ package connection
 
 import (
 	"context"
+	"crypto/sha256"
```

Comment:
Please add to the commit message the reason why this new system has been added.
For example, we should mention that the old `Code Flow` system suffers from some vulnerabilities that PKCE does not suffer from.